### ISSUE TITLE
Focus not being "re Gained" 

### ISSUE THREAD

**Issue by matias-tecnosoul** [EltonFaust/cordova-plugin-multi-player#11]
Hi, I can't find how to regain Focus after losing it after a Whapp audio message being sent (for example). This is what I'm getting in the logs:

_MultiPlayer: RadioPlugin : RADIO STATE - PLAYING...
MultiPlayer: RadioPlayerService : Player state changed. Stopped focus loss
MultiPlayer: RadioPlugin : RADIO STATE - STOPPED FOCUS LOSS..._

Which is perfect AFAIK. The thing is that it just stays there after focus being released by the other app.
thanks for any hints!

**Comment by matias-tecnosoul** [comment]
I've just realized about this option:
_else if (s == 'STOPPED_FOCUS_LOSS') {
            // the media was stopped after other app requested focus_
I'll analyze it and see if I can solve the issue this way...
thanks again for any hints...

**Comment by matias-tecnosoul** [comment]
I could fire a function when _s == 'STOPPED_FOCUS_LOSS'_ but this happens right after losing focus, so it's not really what I need, which is restarting playback (or regaining focus) after the other app releases the audio focus.
Sorry if I'm missing smtg really basic here, it's the first time I'm dealing with audio focus :p
I've found this plugin [Cordova-plugin-audiofocus](https://github.com/Nikunj-Jain/Cordova-plugin-audiofocus) which I'm checking now to see if it can help.

**Comment by matias-tecnosoul** [comment]
Sorry to bump this thread... but could we have a hint on how to deal with audio focus?
do we need a listener once we loose focus?
This is what I'm getting in the logs when I'm trying to send an audio message in Whatsapp or Telegram:

> V MultiPlayer: RadioPlayerService : Player state changed. Stopped focus loss
> V MultiPlayer: Plugin : RADIO STATE - STOPPED FOCUS LOSS...

thanks!

**Comment by EltonFaust** [comment]
Hello,

When you try to record/listen to something in other apps, probably they request the audio focus from android for their app

There are basically 3 types of focus loss:
 - [Loss](https://developer.android.com/reference/android/media/AudioManager#AUDIOFOCUS_LOSS): lose the focus and can't be automatically regained
 - [Loss Transient](https://developer.android.com/reference/android/media/AudioManager#AUDIOFOCUS_LOSS_TRANSIENT): lose the focus, but can probably be regained in a while
 - [Loss Transient Can Duck](https://developer.android.com/reference/android/media/AudioManager#AUDIOFOCUS_LOSS_TRANSIENT_CAN_DUCK): something else started playing but your audio should continue playing in a lower volume

The plugin currently support the **_Can Duck_**, by lowering the volume to 20% once loss the focus and go back to 100% once regained.

The other 2 types are currentlly considered the same in the plugin, just stopping the audio, and sending the **_STOPPED_FOCUS_LOSS_** event, and if this happend, the audio will not be restarted even if a [Regain](https://developer.android.com/reference/android/media/AudioManager#AUDIOFOCUS_GAIN_TRANSIENT) from android is received.

So basically, in the current state of the plugin, is required to call the **_play_** method again to restart playing and regain the audio focus, something that probably would require some kind of user action.

Probably in the future I will add some event when the audio can be regained so you could call automatically the **_play_** action if you wanted, but unfortunately right now doesn't has some workaround...

**Comment by matias-tecnosoul** [comment]
Hi, thanks for the information! 
I see. The ducking works when listening an audio message, and when recording one, the LOSS is triggered and the playing stops.

Thanks a lot for the plugin!
